ideascube/serveradmin/models.py

```diff
@@ -0,0 +1,55 @@
+from django.conf import settings
+from django.db import models
+
+from ideascube.models import JSONField
+
+
+class Setting(models.Model):
+    class Meta:
+        ordering = ['-date']
+
+    namespace = models.CharField(max_length=40)
+    key = models.CharField(max_length=40)
+    value = JSONField()
+    actor = models.ForeignKey(settings.AUTH_USER_MODEL)
+    date = models.DateTimeField(auto_now=True)
+
+    @classmethod
+    def _get_typed(cls, namespace, key, type, default=None):
+        try:
+            setting = Setting.objects.get(namespace=namespace, key=key)
+
+        except Setting.DoesNotExist:
+            if default is not None:
+                return default
+
+            raise
+
+        else:
+            if isinstance(setting.value, type):
+                return setting.value
+
+            if default is not None:
```

Comment:
>  It may be one reason : evolution of the settings.

That is :)
We could also catch the error every where we use call for in-db setting, but for what benefit? Raising a warning? What the user will do with the warning?
And still, if in some place we do want to catch the error externally, we just do not provide a default.